common/cloud_builder.py

```diff
@@ -0,0 +1,261 @@
+"""CloudBuilder executes agents in Cloud Builds."""
+import argparse
+import logging
+import os
+import subprocess
+import tempfile
+import time
+import uuid
+from typing import Any
+
+import google.api_core.client_options
+import googleapiclient.errors
+from google.api_core.exceptions import NotFound
+from google.auth import default
+from google.auth.transport.requests import Request
+from google.cloud import storage
+from googleapiclient.discovery import build as cloud_build
+
+import utils
+from agent.base_agent import BaseAgent
+from results import BuildResult, Result
+
+OF_REPO = 'https://github.com/google/oss-fuzz.git'
+US_CENTRAL_CLIENT_OPTIONS = google.api_core.client_options.ClientOptions(
+    api_endpoint='https://us-central1-cloudbuild.googleapis.com/')
+
+
+class CloudBuilder:
+  """A worker to execute llm-agents workflow in Google Cloud Build, providing a
+  scalable and distributed alternative to local executions:
+  - Request, monitor, and manage Google Cloud Build jobs.
+  - Execute agent in the cloud environment, replicating the local conditions.
+  - Transfer data and results between local and cloud environment.
+  """
+
+  def __init__(self, args: argparse.Namespace) -> None:
+    #TODO(dongge): extra tags.
+    self.credentials, self.project_id = default()
+    assert self.project_id, 'Cloud experiment requires a Google cloud project.'
+    assert hasattr(
+        self.credentials,
+        'refresh'), ('Cloud experiment requires a service account email')
+    assert hasattr(self.credentials, 'service_account_email'), (
+        'Cloud experiment requires a service account email')
+
+    try:
+      # TODO(dongge): Understand why this crashes in local experiments.
```

Comment:
What's the exception that happens?

Reply:
A fresh error:
```python
google.auth.exceptions.RefreshError: ('invalid_scope: Invalid OAuth scope or ID token audience provided.', {'error': 'invalid_scope', 'error_description': 'Invalid OAuth scope or ID token audience provided.'})
```
[Full log](https://paste.googleplex.com/6008336056320000#l=21).

This does not block any experiments because it only happens when requesting cloud builds from a local device, which does not require a refresh.